testing-common/src/main/java/io/opentelemetry/instrumentation/testing/junit/http/HttpClientTests.java

```diff
@@ -0,0 +1,1063 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.testing.junit.http;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.NetTransportValues.IP_TCP;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.junit.jupiter.api.DynamicTest.dynamicTest;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import io.opentelemetry.instrumentation.testing.InstrumentationTestRunner;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.testing.assertj.TraceAssert;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.net.URI;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.function.Executable;
+
+public final class HttpClientTests<REQUEST> {
+
+  public static final Duration CONNECTION_TIMEOUT = Duration.ofSeconds(5);
+  public static final long CONNECTION_TIMEOUT_MS = CONNECTION_TIMEOUT.toMillis();
+  public static final Duration READ_TIMEOUT = Duration.ofSeconds(2);
+  public static final long READ_TIMEOUT_MS = READ_TIMEOUT.toMillis();
+
+  static final String BASIC_AUTH_KEY = "custom-authorization-header";
+  static final String BASIC_AUTH_VAL = "plain text auth token";
+
+  private final InstrumentationTestRunner testRunner;
+  private final HttpClientTestServer server;
+
+  private final HttpClientTestOptions options;
+  private final HttpClientTypeAdapter<REQUEST> clientAdapter;
+
+  public HttpClientTests(
+      InstrumentationTestRunner testRunner,
+      HttpClientTestServer server,
+      HttpClientTestOptions options,
+      HttpClientTypeAdapter<REQUEST> clientAdapter) {
+    this.testRunner = testRunner;
+    this.server = server;
+    this.options = options;
+    this.clientAdapter = clientAdapter;
+  }
+
+  public List<DynamicTest> allList() {
+    return all().collect(Collectors.toList());
+  }
```

Comment:
You don't really need this method, a `@TestFactory` can return a `Stream` and JUnit 5 will work with that.

BTW this might make the IntelliJ problem a bit easier to handle -- if you want to selectively enable tests, you could just drop a 
```java
.filter(test -> test.getDisplayName().equals("aaaaa"))
```
in your actual test class. (Well, I'm not saying that it's a great idea, but it's definitely better than groovy)